cloudfoundry/cf-create-org.sh

```diff
@@ -1,134 +1,124 @@
 #!/bin/bash
+set -e -o pipefail
 
-#set -e
-set -x
+# import common functions
+. "$(dirname "$0")/../lib/common.sh"
 
-if [ "$#" -lt 5 ]; then
-  printf "Usage:\n\n\t\$./cf-create-org.sh <AGENCY_NAME> <IAA_NUMBER> <SYSTEM_NAME> <POP_START> <POP_END> <MANAGER> <MEMORY> <MANAGER_ORIGIN>\n\n"
-  exit 1
-fi
-
-AGENCY_NAME=$1
-IAA_NUMBER=$2
-SYSTEM_NAME=$3
-POP_START=$4
-POP_END=$5
-MANAGER=$6
-MEMORY=$7
-MANAGER_ORIGIN=$8
-
-ORIGIN_FLAG=""
-
-# Checks to see if an origin was set for the manager; if so, this will be used
-# for setting the roles later on.
-if [[ "${MANAGER_ORIGIN}" ]]; then
-  ORIGIN_FLAG="--origin ${MANAGER_ORIGIN}"
-fi
-
-if ! [[ $AGENCY_NAME =~ ^[a-zA-Z0-9]+$ ]]; then
-  echo "AGENCY_NAME must contain only letters and numbers."
-  exit 1
-fi
-
-if ! [[ $IAA_NUMBER =~ ^[a-zA-Z0-9_\.-]+$ ]]; then
-  echo "IAA_NUMBER must contain only letters, numbers, underscores, periods, and hyphens."
-  exit 1
-fi
-
-if ! [[ $SYSTEM_NAME =~ ^[a-zA-Z0-9-]+$ ]]; then
-  echo "SYSTEM_NAME must contain only letters, numbers, and hyphens."
-  exit 1
-fi
-
-if ! [[ $POP_START =~ ^[0-9]{8}$ ]]; then
-  echo "POP_START should be of the format YYYYMMDD."
-  exit 1
-fi
-
-if ! [[ $POP_END =~ ^[0-9]{8}$ ]]; then
-  echo "POP_END should be of the format YYYYMMDD."
-  exit 1
-fi
-
-QUOTA_NAME_BASE="${AGENCY_NAME}_${IAA_NUMBER}_${POP_START}-${POP_END}"
-# uppercase
-QUOTA_NAME_BASE=$(echo "$QUOTA_NAME_BASE" | awk '{print toupper($0)}')
-ORG_NAME="${AGENCY_NAME}-${SYSTEM_NAME}"
-# lowercase
-ORG_NAME=$(echo "$ORG_NAME" | awk '{print tolower($0)}')
+# set some defaults
+MEMORY_QUOTA=8
```

Comment:
It feels somewhat risky to have a default for memory and not make it a required argument?